src/libraries/Microsoft.Bcl.Cryptography/src/System/Security/Cryptography/NetStandardShims.cs

```diff
@@ -102,11 +102,11 @@ internal static bool TryGetHashAndReset(
         {
             int hashSize = hash.AlgorithmName.Name switch
             {
-                nameof(HashAlgorithmName.MD5) => 128 >> 3,
-                nameof(HashAlgorithmName.SHA1) => 160 >> 3,
-                nameof(HashAlgorithmName.SHA256) => 256 >> 3,
-                nameof(HashAlgorithmName.SHA384) => 384 >> 3,
-                nameof(HashAlgorithmName.SHA512) => 512 >> 3,
+                nameof(HashAlgorithmName.MD5) or "HMACMD5" => 128 >> 3,
```

Comment:
There are two ways to fix this.

1. The way that you see in this pull request.
2. Delete the entire switch on the algorithm name. It is only used so we know the length of the hash before we call `GetHashAndReset`, presumably to avoid the allocation.

    However, since this is a .NET Standard shim, we could just take the allocation, and check the length of `destination` against what `actual` is. The current switch only saves an allocation in an error path, which may not be worth optimizing.

Reply:
> we could just take the allocation, and check the length of destination against what actual is. The current switch only saves an allocation in an error path, which may not be worth optimizing.

Yeah, let's go for that.  It saves updating this switch if we end up with SHA3 support here in some theoretical future.